internal/transport/controlbuf.go

```diff
@@ -726,6 +727,16 @@ func (l *loopyWriter) outFlowControlSizeRequestHandler(o *outFlowControlSizeRequ
 func (l *loopyWriter) cleanupStreamHandler(c *cleanupStream) error {
 	c.onWrite()
 	if str, ok := l.estdStreams[c.streamID]; ok {
+		// If the stream is active, allow the return buffer cleanup function to
+		// proceed because processData is not going to execute on this stream to
+		// call wg.Done() anymore.
+		if str.state == active {
+			for !str.itl.isEmpty() {
+				if dataItem, ok := str.itl.dequeue().(*dataFrame); ok && dataItem.wg != nil {
```

Comment:
Good catch. Only issue I see with that is that this would require a interface type assertion for every dequeue to check if it's a `*dataFrame` even when we don't expect it to be. For example, `originateStream` expects a `*headerFrame` in the item list head.

This would also mix the item list linked list abstraction with gRPC-specific stuff (right now it's just a `interface{}` LL), but that's not really a problem since this is all internal stuff (but I thought I should mention it).

Reply:
One idea to work around the abstraction leakage and type assertion: add an `onDequeue` field to `itemNode` that `itemList.dequeue()` calls unconditionally (if non-nil).  I don't think this would add too much overhead.

Reply:
Done.